analyzers/src/SonarAnalyzer.CSharp/Rules/CloudNative/AzureFunctionsLogFailures.cs

```diff
@@ -0,0 +1,189 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class AzureFunctionsLogFailures : SonarDiagnosticAnalyzer
+    {
+        private const string DiagnosticId = "S6423";
+        private const string MessageFormat = "Log caught exceptions via ILogger with LogLevel Warning, Error, or Critical";
+
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        private static readonly ILanguageFacade LanguageFacade = CSharpFacade.Instance;
```

Comment:
I'm interested in `ILanguageFacade.MethodParameterLookup`. I did a `new CSharpMethodParameterLookup` before but found the indirection via `ILanguageFacade` useful to allow for a more flexible evolution later. I also like to place dependency in the constructor (not possible here) or at least at the top of the file. Dependencies should not be scattered around in the file imo.

Reply:
Just remove it and use the `CSharpMethodParameterLookup` directly. The facade is meant to be used only to unify access to stuff for VB and CS rules. Currently, it only adds uneeded level of indirection.
The dependency concern is not an issue for rules, because they are the final users and nothing cannot depend on them. So the dependency overview is needed here.

Reply:
Consider the intrudction of a `SonarCSharpOnlyDiagnosticAnalyzer` that gives equivalent benefits as ` SonarDiagnosticAnalyzer<TSyntaxKind>` does for rules that are supported both by VB.NET and C#.

The Language facade(s) also contain node logic that is useful when you only deal with C#. Then you still can have the debate what makes the intention of the code clearer, using the Language(facade) abstraction or `CSharpMethodParameterLookup` directly.

Reply:
I removed the facade and called `new CSharpMethodParameterLookup.TryGetNonParamsSyntax` directly. Everything else just complicates the solution without much benefit. The `TryGetNonParamsSyntax` method could also be turned into an extension method for C#. This would be more in line with how we made language-neutral implementations available in the specific language recently.